test/components/popup.spec.tsx

```diff
@@ -28,68 +50,82 @@ describe('<Popup />', function() {
 
     describe('The popup user', function() {
         it('clicks on the parent and the popup opens and closes after another click', async function() {
-            const {select, waitForDom} = clientRenderer.render(<PopupDemo />);
+            const {result, waitForDom} = clientRenderer.render(<PopupDemo />);
+
+            let popupDemo = new PopupDemoTestDriver(result as PopupDemo);
 
             await waitForDom(() => {
-                expect(select(demoContainer)).to.be.present();
-                expect(select(demoContainer, portalId)).to.be.absent();
+                expect(popupDemo.container).to.be.present();
+                expect(popupDemo.isPopupPresent).to.be.false;
             });
-            select<HTMLDivElement>(demoContainer)!.click();
-            await waitForDom(() => expect(bodySelect(portalId)).to.be.present());
-            select<HTMLDivElement>(demoContainer)!.click();
-            return waitForDom(() => expect(bodySelect(portalId)).to.be.absent());
+
+            (popupDemo.container as HTMLDivElement).click();
+            popupDemo = new PopupDemoTestDriver(result as PopupDemo);
```

Comment:
This is a bit weird. It is happening because the portal is not open when the popup demo renders.
I think a possible solution for this is to manage the clicking on the demo container inside the demo driver and there updating the popupDriver after clicking.
Either way you don't need the third creation of the popup demo driver.